src/index.js

```diff
@@ -35,8 +35,15 @@ export async function run() {
       await execShellCommand('pacman -Sy --noconfirm tmate');
       tmateExecutable = 'CHERE_INVOKING=1 tmate'
     } else {
-      await execShellCommand(optionalSudoPrefix + 'apt-get update');
-      await execShellCommand(optionalSudoPrefix + 'apt-get install -y openssh-client xz-utils');
+      const distro = await getLinuxDistro();
+      core.debug("linux distro: [" + distro + "]");
+      if (distro === "alpine") {
+        // for set -e workaround, we need to install bash because alpine doesn't have it
+        await execShellCommand(optionalSudoPrefix + 'apk add openssh-client xz bash');
```

Comment:
is sudo a thing on alpine?

Reply:
@mxschmitt hmm, I think alpine container image (main use case of alpine) does not have it.

I try to fix it, but things bit complex because `sudo` will be required for invoking `apk add sudo` if user want to use `sudo`. so I think 2 options 

1. if sudo option is set for alpine, raise error like `sudo option is not supported for alpine`
2. invoking `apk` without `optionalSudoPrefix` for installing `sudo` on `alpine`

personally I prefer option 1 because alpine's main use case is as container image, which should not require `sudo`. also user have an option to install sudo for their alpine based image.

how do you think?

Reply:
with https://github.com/mxschmitt/action-tmate/pull/100/commits/c93ade251d8628e42feb3d4178e1c1532c444f66 I implement option 1.

Reply:
I think another option is enable sudo option but don't install sudo command by the action (cancel https://github.com/mxschmitt/action-tmate/pull/100/commits/c93ade251d8628e42feb3d4178e1c1532c444f66). 

some user may use non-root user for their alpine based image.

in this case, user can use the action if they install sudo on their image creation.

Reply:
An even better method would be to use https://nodejs.org/api/process.html#processgeteuid to determine whether `sudo` is even needed. I bet that we're running as root in the Alpine container already.

Reply:
Oh, never mind, I just realized that we're not using `sudo` by default. Sorry for the noise.